### DISCUSSION TITLE
Can't start with different ports

### DISCUSSION THREAD

**Discussion by HeikoBo** [wolveix/satisfactory-server#167] [Q&A] [answered]
Hello,

I tried to create 2 Satis docker instances on Unraid. But it seem that it is not possible to change the port.
I tried it and changed the 3 ports but in the logfiles I see that steam tries to do something on the default port.

Have anyone tried it out to change the default ports? Maybe I have overseeen something.

Thanks,
Heiko

**Comment by wolveix** [comment, marked as answer]
Hi @HeikoBo. I'd recommend just remapping rather than reconfiguring, which you can do in your Docker Compose file or run command.

In the README for the repo, you'll see that there are three environment variables for changing the internal ports.

**Reply by HeikoBo** [reply]
Hello @wolveix. I did that but it seems that this is not working. I Use Unraid and I have one Docker with the default settings and everythink works fine. For the second Docker I increased the 3 Host-Ports by one. My run command looks that way:

docker run
  -d
  --name='satisfactory_internal'
  --net='bridge'
  -e TZ="Europe/Berlin"
  -e HOST_OS="Unraid"
  -e HOST_HOSTNAME="Unraid"
  -e HOST_CONTAINERNAME="satisfactory_internal"
  -e 'MAXPLAYERS'='4'
  -e 'PGID'='1000'
  -e 'PUID'='1000'
  -e 'STEAMBETA'='false'
  -l net.unraid.docker.managed=dockerman
  -l net.unraid.docker.icon='https://img2.storyblok.com/fit-in/0x200/filters:format(webp)/f/110098/268x268/d1ebbafe03/logo.png'
  -p '7778:7777/udp'
  -p '15001:15000/udp'
  -p '15778:15777/udp'
  -v '/mnt/user/appdata/satisfactory_internal':'/config':'rw' 'wolveix/satisfactory-server'

I added one Server with Port 15777 and the other one with port 15778 
And now the strange think is, that I see on both entries the servername of the Docker with Port 15777.
I am able to login to this server but both entries directs me to the server with port 15777.

I'm confused :-)

Any help would be great.

Thanks,
Heiko

**Reply by HeikoBo** [reply]
Hmm. Maybe there is some bug in the game/system. 

- Lets say I entered a new Server in Satis to connect to a docker on port 15778. (Server 2) Within the docker satis is running on 15777 (default). 
- Now a connection is established and the server throws back the other internal ports for satis which are also default. In this case satis is not connecting to the Server on port 15778 (Server 2) but to the server 15777 which is also running on the same host in the other docker. (Server 1)
- This means it would not be possible run satis with same internal ports in docker. Satis Server has to start also with differnt ports directly?

Just some thoughts..

**Reply by wolveix** [reply]
Hey @HeikoBo. I'm unable to test right now, but if memory serves me right, you need to also change the internal server ports for the second server as it fails to accept traffic flowing through ports that it's not configured for due to a second state check.

Try setting the three environment variables, and make sure that the port mapping in Docker is 1:1 with the new ports.

**Reply by HeikoBo** [reply]
Yes, It is not enough to just change the hosting ports. The internal ports must also be changed. But it seems, that the startup script of the Satisserver is still using the default ports. Maybe it is a bug of the docker image? 

Maybe the startup script have to get the enviroment variables or there shout be an enviroment variable to change the startup script line of the server, so we you can enter the ports manually?

**Reply by wolveix** [reply]
@HeikoBo as you can see from https://github.com/wolveix/satisfactory-server/blob/main/run.sh#L142, the environment variables are definitely being passed to the game server. Could you send your Docker run/Docker Compose command/file please?
